src/emoji/Emoji.js

```diff
@@ -1,9 +1,37 @@
 /* @flow strict-local */
-import { createIconSet } from 'react-native-vector-icons';
-import { nameToEmojiMap } from './data';
+import React, { PureComponent } from 'react';
 
-/* $FlowFixMe: `nameToEmojiMap` is mistyped upstream; elements of
-  `glyphMap` may be either `number` or `string`. */
-const Icon = createIconSet(nameToEmojiMap, 'AppleColorEmoji');
+import type { ImageEmojiType, Dispatch } from '../types';
+import { connect } from '../react-redux';
+import UnicodeEmoji from './UnicodeEmoji';
+import ImageEmoji from './ImageEmoji';
+import { getActiveImageEmojiByName } from './emojiSelectors';
 
-export default Icon;
+type SelectorProps = {|
+  imageEmoji: ImageEmojiType | void,
+|};
+
+type Props = {|
+  name: string,
+
+  dispatch: Dispatch,
+  ...SelectorProps,
+|};
+
+type ComponentProps = {|
+  name: string,
+|};
```

Comment:
Let's avoid this redundancy -- there's no need for it. Search for `SelectorProps` to see examples.